cmd/query/app/handler_options.go

```diff
@@ -64,7 +64,8 @@ func (handlerOptions) QueryLookbackDuration(queryLookbackDuration time.Duration)
 // Tracer creates a HandlerOption that initializes OpenTracing tracer
 func (handlerOptions) Tracer(tracer jtracer.JTracer) HandlerOption {
 	return func(apiHandler *APIHandler) {
-		apiHandler.tracer = tracer
+		apiHandler.tracer.OT = tracer.OT
+		apiHandler.tracer.OTEL = tracer.OTEL
```

Comment:
why does apiHandler need both? If it passes tracer further downstream, it should pass JTracer. If it uses it directly for instrumentation, then it should only take one of these.